[Enhancement] delete some compaction configurations (#5363)

xiaoyong-z opened the pull request:
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #5363 

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->


Modification details:
```txt
# 7. unused configurations, remove
CONF_mInt32(base_compaction_write_mbytes_per_sec, "5");

# 12. unused configurations, remove
CONF_mInt64(cumulative_compaction_budgeted_bytes, "104857600");

# 25. use cumulative_compaction_num_threads_per_disk to replace it, remove
CONF_mInt32(max_cumulative_compaction_task, "-1");

# 26. use base_compaction_num_threads_per_disk to replace it, remove
CONF_mInt32(max_base_compaction_task, "1");

# 30. remove precent limit
CONF_Int32(compaction_max_memory_limit_percent, "100");
```

imay commented:
@xiaoyong-z 
I see you did many things in one PR. 
I suggest that you split this PR into many.
One PR do only one thing. For example, you can delete unused configuration in on PR, change other things in another PR.
And I see that you changed configuration name and default value, this is a danger behavior. This will cause old configuration in configure file useless which can cause user product environment problems

imay commented on be/src/common/config.h:
```diff
@@ -288,9 +286,7 @@ CONF_mInt64(min_compaction_failure_interval_sec, "120"); // 2 min
-CONF_mInt32(base_compaction_trace_threshold, "120");
-CONF_mInt32(cumulative_compaction_trace_threshold, "60");
-CONF_mInt32(update_compaction_trace_threshold, "20");
+CONF_mInt32(compaction_trace_threshold, "60");
```
Better change this in another PR

imay commented on be/src/storage/compaction_scheduler.cpp:
```diff
@@ -90,15 +90,8 @@ bool CompactionScheduler::_can_do_compaction_task(Tablet* tablet, CompactionTask
-            compaction_task->compaction_type());
     DataDir* data_dir = tablet->data_dir();
     if (compaction_task->compaction_type() == CUMULATIVE_COMPACTION) {
-        if (config::max_cumulative_compaction_task >= 0 && task_num >= config::max_cumulative_compaction_task) {
```
why remove this check?

xiaoyong-z replied:
The function of `config::max_cumulative_compaction_task)` is somewhat duplicated by the function of `config::cumulative_compaction_num_threads_per_disk`, lator we will check task_num less than `config::cumulative_compaction_num_threads_per_disk`. 
Same reason for `config::base_compaction_num_threads_per_disk`.

On the other hand, we already have the `max_compaction_task_num` to make sure the overall task_num will not exceed the limit, i think there is no need to set another maximum task limit for each type of compaction.

imay replied:
@chaoyli should double-check this

chaoyli commented on be/src/common/config.h:
```diff
@@ -462,7 +457,6 @@ CONF_Int64(load_process_max_memory_limit_bytes, "107374182400"); // 100GB
 CONF_Int32(load_process_max_memory_limit_percent, "30");         // 30%
 CONF_Bool(enable_new_load_on_memory_limit_exceeded, "false");
 CONF_Int64(compaction_max_memory_limit, "-1");
-CONF_Int32(compaction_max_memory_limit_percent, "100");
```
The compaction_max_memory_limit_percent is used to limit the percent used.

trueeyu replied:
Don't remove it, we will change it to one small value if the compaction task support better scheduler

xiaoyong-z replied:
okay, i will reserve it.

trueeyu commented on be/src/runtime/exec_env.cpp:
```diff
@@ -93,10 +92,8 @@ static int64_t calc_max_compaction_memory(int64_t process_mem_limit) {
-    }
-    return std::min<int64_t>(limit, process_mem_limit * percent / 100);
+
+    return std::min<int64_t>(limit, process_mem_limit);
```
current the config::compaction_memory_limit_per_worker is only used for change the chunk size, unable to limit the memory usage of one task.

xiaoyong-z committed:
[Enhancement] delete some compaction configurations (#5363)

    The following configurations are deleted:

        CONF_mInt32(base_compaction_write_mbytes_per_sec, "5");

        CONF_mInt64(cumulative_compaction_budgeted_bytes, "104857600");

        CONF_mInt32(max_cumulative_compaction_task, "-1");

        CONF_mInt32(max_base_compaction_task, "1");

Signed-off-by: xyz <a997647204@gmail.com>
